File: kotlin/src/main/kotlin/problems/problem1/two_sum.kt
package problems.problem1


class Solution {

    fun twoSum(nums: IntArray, target: Int): IntArray {
        val h = hashMapOf<Int, Int>()
        for ((i, v) in nums.withIndex()) {
            val j = h[target - v]
            if (j != null) {
                return intArrayOf(i, j)
            }
            h[v] = i
        }
        throw IllegalArgumentException()
    }
}

File: go/explore/august2021/day2/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	m := map[int]int{}
	for i, num := range nums {
		m[num] = i
	}
	for i, num := range nums {
		j, ok := m[target-num]
		if ok && j != i {
			return []int{i, j}
		}
	}
	return []int{}
}
